modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java

```diff
@@ -346,12 +344,6 @@ private boolean enrollDevice(Document request) throws WindowsDeviceEnrolmentExce
                 }
                 Device generateDevice = generateDevice(DeviceManagementConstants.MobileDeviceTypes.
                         MOBILE_DEVICE_TYPE_WINDOWS, devID, modVersion, imsi, imei, devMan, devMod, user);
-                DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(syncmlHeader.getSource()
-                        .getLocURI());
-               // Device existingDevice = WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier);
-                if (WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier) != null) {
-                    WindowsAPIUtils.getDeviceManagementService().modifyEnrollment(generateDevice);
-                }
                 status = WindowsAPIUtils.getDeviceManagementService().enrollDevice(generateDevice);
```

Comment:
possible NPE scenario if we do not get the service here. better checking it first and invoke .enrollDevice